### PR TITLE
Va 102 mobile conversation unread on load

### PR DESCRIPTION
**Pull request by DaceyTom2**
### Description
Mobile will not automatically read messages on load

### Note
Loading chat window was moved to a new component
isMobile added to state to track if we are in a mobile version of the app
isMobile is bypassed in testing to keep tests consistent no matter the view size tests are run in

### Test
Open app in Mobile
Messages should be unread
Click chat and minimise
Messages should be read

Open app in Desktop
Messages should be read

### CODE REVIEW

**Comment by TishamIslam** [comment]
Broke the job bank window on desktop view, not sure why (no console message).

File: src/components/molecules/InboxItem.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
     @keyup.enter="selectInboxItem"
     :class="[
       inboxItem.selected
-        ? 'bg-blue-selected rounded-none'
+        ? 'md:bg-blue-selected md:rounded-none'
```
Comment (TishamIslam) [inline src/components/molecules/InboxItem.vue:8]: Desktop view now uses the sm (>= 640 px) breakpoint, not the md (>= 768 px) breakpoint.

File: src/store/modules/inbox.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ const getters = {
       //dayRead will be unread if no value provided. The design currently accounts for only the weekday to be displayed
       let dayRead;
       const lastReadDate = item.lastRead ? new Date(item.lastRead) : null;
+      console.log(lastReadDate);
```
Comment (TishamIslam) [inline src/store/modules/inbox.js:23]: Should be removed.

**Review by TishamIslam — CHANGES_REQUESTED** [review]
Looked into it, the reason the Job Bank tab breaks on desktop is the `v-show` here: 
image

Changing it to something like this (similarly to the ConversationWindow) and removing the v-show, should suffice:
```Vue
<div :class="[
      'w-full h-full sm:flex flex-col',
      isMobileDrawerOpen ? 'flex' : 'hidden',
    ]" >
```
    
Also some minor code changes below.

**Comment by DaceyTom2** [comment]
@TishamIslam I don't believe we need the v-show at all for message list. The parent component handles it for us.

File: src/store/modules/inbox.js (hunk 2 above)
Comment (DaceyTom2) [reply src/store/modules/inbox.js]: Good catch... Oops...

File: src/components/molecules/InboxItem.vue (hunk 1 above)
Comment (DaceyTom2) [reply src/components/molecules/InboxItem.vue]: Thank you, I didn't realise this had changed

**Comment by TishamIslam** [comment]
The v-show/equivalent logic makes it so that the messages can't be accessed via tabbing or screenreaders when you exit the drawer.

**Comment by DaceyTom2** [comment]
I followed the solution but I had to apply sm:hidden and md:flex on drawer closed since it was hiding message list

**Comment by TishamIslam** [comment]
Seems like in between sm and md breakpoints (640-768px) the message list disappears
image

**Comment by TishamIslam** [comment]
Try replacing `sm:hidden md:flex` with `hidden sm:flex`

**Comment by DaceyTom2** [comment]
@TishamIslam it looks like the breakpoints are now xs and sm. for mobile to desktop. It is also worth noting sometimes tailwind isn't always reacting to the screen size so try refreshing the page if it doesn't show

**Review by nibivi77 — CHANGES_REQUESTED** [review]
On mobile initial load, there's a split second where the VA inbox item is set to read state until it goes back to unread. Would there be a way handle that?

**Comment by DaceyTom2** [comment]
@nibivi77 The flickering isn't related to this PR but I added in a fix for sanity sake

**Comment by nibivi77** [comment]
> @nibivi77 The flickering isn't related to this PR but I added in a fix for sanity sake

Ah Sorry I did not realise.

**Review by nibivi77 — APPROVED** [review]
This seems to do the job. Looks good to me Tom!

**Review by TishamIslam — APPROVED** [review]
Everything seems to be in order :)

**Pull request merged**